.github/workflows/main.yml

```diff
@@ -50,27 +50,41 @@ jobs:
         run: make install_cli_deps
       - name: generate protobufs, RPC server, RPC client and mocks
         run: make protogen_local && make mockgen && make generate_rpc_openapi
-      - name: run all tests
-        run: make test_all_with_json
+      - name: Create coverage report and run tests
+        # Not utilizing makefile target here to make use of pipefail bash feature.
+        run: |
+          set -euo pipefail
+          make test_all_with_json_coverage
```

Comment:
Actually, @Olshansk can we bring utilizing the CLI back here? I tried to make this work with `Makefile`, but the reason job succeeded is because `set -euo pipefail` didn't propagate into makefile target. Makefile doesn't utilize bash shell. The test still failed.

Reply:
<img width="1099" alt="image" src="https://user-images.githubusercontent.com/4950477/204882193-3b70723a-0d89-4df5-a1a3-075255b2736f.png">

What happens is due to the fact we are utilizing `tee`, the `exit(1)` from `go test` never forces the CI job to fail:
```
test_all_with_json_coverage: generate_rpc_openapi # generate_mocks
	go test -p 1 -json ./... -covermode=count -coverprofile=coverage.out | tee test_results.json | jq
```
Due to the fact make targets do not run in bash shell, hence it does not recognize `set -euo pipefail` (the actual setting that forces non-zero exit code to propagate to the CLI execution down the pipe). 

I've tried to make this work on my local PR with different ways, but decided it is best to use go CLI directly, and I put this comment for that reason: `# Not utilizing makefile target here to make use of pipefail bash feature.`

Reply:
[`6742174` (#285)](https://github.com/pokt-network/pocket/pull/285/commits/67421745f5810449ce3871eada61290c86bb4369)

Rerunning now